services/skus/storage/repository/issuer_test.go

```diff
@@ -0,0 +1,284 @@
+		})
+	}
+}
+
+func TestIssuer_GetByPubKey(t *testing.T) {
+	dbi, err := setupDBI()
+	must.Equal(t, nil, err)
+
+	t.Cleanup(func() {
+		_, _ = dbi.Exec("TRUNCATE_TABLE order_cred_issuers;")
+	})
+
+	type tcGiven struct {
+		pubKey string
+		mid    string
+		pkey   string
+	}
+
+	type tcExpected struct {
+		result *model.Issuer
+		err    error
+	}
+
+	type testCase struct {
+		name  string
+		given tcGiven
+		exp   tcExpected
+	}
+
+	tests := []testCase{
+		{
+			name: "not_found",
+			given: tcGiven{
+				pubKey: "not_found",
+			},
+			exp: tcExpected{
+				err: model.ErrIssuerNotFound,
+			},
+		},
+
+		{
+			name: "result_1",
+			given: tcGiven{
+				pubKey: "public_key",
+				mid:    "merch_id",
+				pkey:   "public_key",
+			},
+			exp: tcExpected{
+				result: &model.Issuer{
+					MerchantID: "merch_id",
+					PublicKey:  "public_key",
+				},
+			},
+		},
+	}
+
+	repo := repository.NewIssuer()
+
+	for i := range tests {
+		tc := tests[i]
+
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+
+			tx, err := dbi.BeginTxx(ctx, &sql.TxOptions{Isolation: sql.LevelReadUncommitted})
+			must.Equal(t, nil, err)
+
+			t.Cleanup(func() { _ = tx.Rollback() })
+
+			if tc.given.pkey != "" {
+				err := seedIssuerForTest(ctx, tx, tc.given.mid, tc.given.pkey)
+				must.Equal(t, nil, err)
+			}
+
+			act, err := repo.GetByPubKey(ctx, tx, tc.given.pubKey)
+			must.Equal(t, true, errors.Is(err, tc.exp.err))
+
+			if tc.exp.err != nil {
+				return
+			}
+
+			should.Equal(t, tc.exp.result.MerchantID, act.MerchantID)
+			should.Equal(t, tc.exp.result.PublicKey, act.PublicKey)
+		})
+	}
+}
+
+func TestIssuer_Create(t *testing.T) {
+	dbi, err := setupDBI()
+	must.Equal(t, nil, err)
+
+	t.Cleanup(func() {
+		_, _ = dbi.Exec("TRUNCATE_TABLE order_cred_issuers;")
+	})
+
+	type tcGiven struct {
+		req model.IssuerNew
+	}
+
+	type tcExpected struct {
+		result *model.Issuer
+		err    error
+	}
+
+	type testCase struct {
+		name  string
+		given tcGiven
+		exp   tcExpected
+	}
+
+	tests := []testCase{
+		{
+			name: "result_1",
+			given: tcGiven{
+				req: model.IssuerNew{
+					MerchantID: "merch_id_1",
+					PublicKey:  "public_key_1",
+				},
+			},
+			exp: tcExpected{
+				result: &model.Issuer{
+					MerchantID: "merch_id_1",
+					PublicKey:  "public_key_1",
+				},
+			},
+		},
+
+		{
+			name: "result_2",
+			given: tcGiven{
+				req: model.IssuerNew{
+					MerchantID: "merch_id_2",
+					PublicKey:  "public_key_2",
+				},
+			},
+			exp: tcExpected{
+				result: &model.Issuer{
+					MerchantID: "merch_id_2",
+					PublicKey:  "public_key_2",
+				},
+			},
+		},
+	}
+
+	repo := repository.NewIssuer()
+
+	for i := range tests {
+		tc := tests[i]
+
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+
+			tx, err := dbi.BeginTxx(ctx, &sql.TxOptions{Isolation: sql.LevelReadUncommitted})
+			must.Equal(t, nil, err)
+
+			t.Cleanup(func() { _ = tx.Rollback() })
+
+			act1, err := repo.Create(ctx, tx, tc.given.req)
```

Comment:
I am not sure about the using act as a variable name as - 
* It refers to a step/phase in a test
* It reduces the self documentation of the tests i.e. there is no indication of what is coming back from the Create call

SUT/DOC maybe a more appropriate terms given this is direct output of the SUT but again I don't think this adds much value over something more descriptive such as issuer.

Reply:
haha, I see that is confusing. I thought it was trying to imply the `act1.MerchantID` i.e. `MerchantID` in the assertion was the actual. yes, actual works if this is the value being asserted